diff --git a/figshare/graphql/v2_resolvers/es_search/utils.py b/figshare/graphql/v2_resolvers/es_search/utils.py
index 8747ad4846..168416f1d5 100644
--- a/figshare/graphql/v2_resolvers/es_search/utils.py
+++ b/figshare/graphql/v2_resolvers/es_search/utils.py
@@ -416,6 +416,9 @@ def facets_mapping(aggs_facets, aggregations):
         if facet_type == IdentifierType.BASE:
             has_more = aggregations[facet]["_aggs"]["sum_other_doc_count"] != 0
             results_facets[facet] = map_base_facet(facet, aggs_facets, buckets, has_more)
+            results_facets[facet]["elements"] = sorted(
+                results_facets[facet]["elements"], key=lambda x: (-x["count"], x["name"])
+            )
         else:
             results_facets[facet] = map_date_facet(buckets)
 
